graphics/macgui/macwindow.cpp

```diff
@@ -173,43 +180,79 @@ static void drawPixelInverted(int x, int y, int color, void *data) {
+	int height = g->h;
+	// We draw rect with outer kColorGreen2 and inner kColorGreen, so on 2 passes we cut out
+	// scene by external shape of the border
+	g->clear(kColorGreen2);
+	g->fillRect(Common::Rect(sz, sz, width - sz, height - sz), kColorGreen);
+}
+
+void MacWindow::drawBorderFromSurface(ManagedSurface *g) {
+	g->clear(kColorGreen2);
+	Common::Rect inside = _innerDims;
+	inside.moveTo(_macBorder.getOffset(kBorderOffsetLeft), _macBorder.getOffset(kBorderOffsetTop));
+	g->fillRect(inside, kColorGreen);
+
+	_macBorder.blitBorderInto(_borderSurface, _active);
+}
+
+void MacWindow::drawSimpleBorder(ManagedSurface *g) {
+
 	bool active = _active, scrollable = _scrollable, closeable = _active, drawTitle = !_title.empty();
 	const int size = kBorderWidth;
 	int x = 0;
 	int y = 0;
 	int width = _borderSurface.w;
 	int height = _borderSurface.h;
-	Graphics::ManagedSurface *g = &_borderSurface;
 
-	// We draw rect with outer kColorGreen2 and inner kColorGreen, so on 2 passes we cut out
-	// scene by external shape of the border
-	int sz = kBorderWidth / 2;
-	g->clear(kColorGreen2);
-	g->fillRect(Common::Rect(sz, sz, width - sz, height - sz), kColorGreen);
+	prepareBorderSurface(g);
+
+	drawBox(g, x, y, size, size);
+	drawBox(g, x + width - size - 1, y, size, size);
+	drawBox(g, x + width - size - 1, y + height - size - 1, size, size);
+	drawBox(g, x, y + height - size - 1, size, size);
+	drawBox(g, x + size, y + 2, width - 2 * size - 1, size - 4);
+	drawBox(g, x + size, y + height - size + 1, width - 2 * size - 1, size - 4);
+	drawBox(g, x + 2, y + size, size - 4, height - 2 * size - 1);
+	drawBox(g, x + width - size + 1, y + size, size - 4, height - 2 * size - 1);
+
 
-	drawBox(g, x,                    y,                     size,                 size);
-	drawBox(g, x + width - size - 1, y,                     size,                 size);
-	drawBox(g, x + width - size - 1, y + height - size - 1, size,                 size);
-	drawBox(g, x,                    y + height - size - 1, size,                 size);
-	drawBox(g, x + size,             y + 2,                 width - 2 * size - 1, size - 4);
-	drawBox(g, x + size,             y + height - size + 1, width - 2 * size - 1, size - 4);
-	drawBox(g, x + 2,                y + size,              size - 4,             height - 2 * size - 1);
-	drawBox(g, x + width - size + 1, y + size,              size - 4,             height - 2 * size - 1);
 
 	if (active) {
-		fillRect(g, x + size, y + 5,           width - 2 * size - 1, 8, kColorBlack);
+		fillRect(g, x + size, y + 5, width - 2 * size - 1, 8, kColorBlack);
 		fillRect(g, x + size, y + height - 13, width - 2 * size - 1, 8, kColorBlack);
-		fillRect(g, x + 5,    y + size,        8,                    height - 2 * size - 1, kColorBlack);
+		fillRect(g, x + 5, y + size, 8, height - 2 * size - 1, kColorBlack);
 		if (!scrollable) {
 			fillRect(g, x + width - 13, y + size, 8, height - 2 * size - 1, kColorBlack);
-		} else {
+		}
+		else {
```

Comment:
This is not in accordance with our code formatting conventions. It should be '} else {'